### PR TITLE
initial elders page.

### LINKED ISSUE #28: Kubernetes Elders Council - proposal
**Problem**:  As the Kubernetes project has pivoted from a Company run open source project to a Foundation owned and community inclusive project, there have been growing pains.  The perceptions of Googler who have more structure and visibility into most of the operational aspects of the projects differ greatly from the perspectives of most of the community members - particularly new project contributors and interested project contributors.  Google intentionally chose not to have a BDFL for all the down sides that has, but a confused “power vacuum” has evolved.  The organic structure appears to a newcomer to be opaque, relationship based and full of cronyism.  There is no clear starting point progression in the project or escalation for conflict resolution etc.  Long standing (primarily Google employed) contributors feel that newcomers are all focused on specific features and aren’t stepping up to contribute to the more material and growing backlog of housekeeping, management, and organizational health tasks.

**Goals** :  Establish clear cultural, technical, and team leadership as well as act as an escalation point with a “last say” on any contentious issues.   

**Requirements**: This group should have a long standing history with the project,  good people management skills, strong technical vision for the project and empathy for the user and FOSS developer perspective.  

**Instantiation**:   The first seed incarnation of the Elders will be voted in as a slate of 3 Elders with a 2 year term (ending August 2018) through the Community Meeting and on the mailing list kubernetes-dev@.  During the first year, the Elders will add 2 members (with a term ending August 2017) to their group for a total of 5 members.

**Refreshes**:  In August each year subsequent the community will vote on the seats which term out.  All positions voted on will have a 2 year term.  The voting will be handled using Condorcet voting and a service such as [CIVS](http://civs.cs.cornell.edu/).  Voter eligibility must be defined, and will require registration during a public registration period.  Nominations for candidates will be accepted for one week prior to the voting period.   

**Resignations**:  In the event that an elder resigns, the remaining elders will appoint a successor to serve until the next scheduled vote.

**Known Issues**:  Elder nomination process. Voter eligibility.

### PR DESCRIPTION
**Pull request by sarahnovotny**
Final Draft -- fixes #28

### CODE REVIEW

**Comment by sarahnovotny** [comment]
I signed it!

File: elders.md [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+# Refreshes:
+In August each year subsequent the community will vote on the seats which term out.  All positions voted on will have a 2 year term.  The voting will be handled using Condorcet voting and a service such as CIVS.  The vote will not be public, but will require standing as an Active Contributor as defined in the [Community Governance documentation](https://github.com/kubernetes/community/blob/master/governance.md).  Nominations for candidates will be accepted for one week prior to the voting period. Incumbents will be automatically nominated unless they opt-out of reelection. Any newly elected member should not increase representation on the board above 2 people per company. In the case where this invariant would be violated, the candidate(s) with the next highest number of votes will be elected.
```
Comment (davidopp) [inline elders.md]: August is a popular vacation month, maybe better to make it September.

**Comment by bgrant0607** [comment]
cc @thockin @smarterclayton

**Comment by bgrant0607** [comment]
Thanks for working on this @sarahnovotny.

However, I think the situation has evolved since this idea originated last spring.

We have an emergent project organization, which is now documented here (though some details are still preliminary or TBD):
https://github.com/kubernetes/community/blob/master/governance.md

We have been using the top-level project approvers (in kubernetes/OWNERS) for approving new top-level directories, sponsoring new repositories, approving new maintainers, and so on. These senior, long-standing project members who have been providing technical governance. The example escalations from the issue (https://github.com/kubernetes/community/issues/28#issuecomment-235977292) could be resolved by this group. I think it's important that people making technical decisions have an explicit relationship to technical contribution to the project and that the group is established in a similar way to other positions of technical oversight (maintainers, approvers, reviewers). We need processes for adding/removing people to/from the group, but it shouldn't be a fixed size (the project is growing), there shouldn't be term limits (why would we kick off the people most knowledgeable about the project?), and the decisions should be made by existing members.

What I think the project is missing is management that would exist if everyone worked at a single company:

* The project needs a manager -- someone to think about and improve the structure of the project, processes, values, etc. A few people have been working on this part time, but it needs more and more consistent attention given the rate of growth of the project. The project became unmanageable about the time the original Elders proposal was made (~April-May 2016 IIRC). 

* A group of leaders (perhaps managers and tech leads of top 10 teams of engineers working full-time on the project?) needs to meet with some regularity and needs to work together to move the project forward. This isn't about technical escalation, scope decisions, or design reviews. Nor should it be about the roadmap planning performed by the PM group. Example issues to resolve include staffing gaps (engineering, docs, test, release, ...), effort gaps (tragedy of the commons), expertise mismatches, priority conflicts, personnel conflicts, oversight of other arms of the project (e.g., SIGs, PMs).

**Comment by bgrant0607** [comment]
cc @shtatfeld

**Comment by sarahnovotny** [comment]
cc @brendandburns

**Comment by smarterclayton** [comment]
I agree with Brian's comments - I think that direction would address the
same problems with a firmer structure.

**Comment by bgrant0607** [comment]
Another idea: contributor and user (developers can choose which category best fits them) feedback sessions scheduled on a semiannual basis.

**Comment by brendandburns** [comment]
@bgrant0607 @smarterclayton 

## tl; dr;

I think we have a good process for individual features, and even for individual releases, but we really need a committee and process for higher-level, longer term vision and direction, as well as higher level project structure and organization.

It's my hope that the elders would fill that gap. But if we'd rather fill it with owners, that seems ok, but we need to explicitly call out these roles in the governance doc, and set up the process for doing this kind of management.

Additionally, I think that using the elders process Sarah identifies for choosing OWNERS still probably makes sense as a starting point for those discussions.

## Details
Let's separate where we write things up (`governance.md` or `elders.md` I don't really care) from what we need.

Even if we assume that top level OWNER == elder, which may or may not be what we want, as Brian mentions, we need a process for managing the people who are in that list.

It seems to me that the elder's proposal put forth by Sarah is a pretty reasonable first draft for how that is established.

I don't see any mention of term limits, just that there are terms. I think that this makes sense as well, since I don't think top level OWNER == member for life, we should be constantly refreshing and replacing. Put another way, I think it's 100% clear that the list of top level owners should be roughly constant, not monotonically increasing.

In addition to Brian's comments about management for resourcing, and project leadership (both of which are very important needs) I really think we _do_ need to have a small group of people who steward and shape the direction of the project, as well as large-scale organizational decisions that have broad project wide implications.

The previous community meeting discussions around feature process and how they were decided and brought across the project I think illustrate the need to have a process for how such a forward looking decision ("we need to institute a feature planning process") are made so that we can point back and say: "the project owners made this decision here"

An additional example is the decision to break up the repository. There is effectively no group of people who makes the decision that splitting up the repository is something that we should do. We have sort of muddled along, but I think having a crisp decision making body for this will help speed things up and provide accountability and "disagree and commit"

I actually think we do also want to have the top level OWNERS/elders be responsible for high-level roadmap, independent of the PM roadmap process. The PM process is good for the 3-6 month timeframe.

However, we effectively do not have anyone who is tasked with at a high level setting the direction and identity for Kubernetes (or even it's top level goals) for a longer 12-18 month time

--brendan

**Comment by bgrant0607** [comment]
See also https://github.com/kubernetes/community/issues/277

**Comment by bgrant0607** [comment]
@brendandburns Quick comment: The role you describe is totally different from the scope proposed for elders up to this point.

**Comment by bgrant0607** [comment]
@brendandburns Longer response.

I think process issues, like the features process, are very different from technical ones, like supporting multiple repositories. Between your feedback and mine, it seems evident that we have multiple different roles that need to be filled.

I agree that some kind of regular owners meeting makes sense.

As for high-level technical direction and identity, that sounds like it should be owners.

**Comment by craigmcl** [comment]
FWIW, I am strongly in favor of Brian and Clayton's position re significantly bolstering the OWNERS process vs creating a net new voted position.  I also agree with Brendan's position that establish a vendor neutral framework that maintains perspective on the long term aspirations of the project (primarily scope and extensibility) is going to be vital to its long term health.  I think this is something that we need to grow out over time vs reflexively create.

In my mind the logical starting point is helping to create structure and clarity around the OWNERS model and processes, put some effective project management resource into the community.  If all goes well we then gradually refine and strengthen the OWNERS model to be a include a promoted and more focused leadership group.  

Today I am fearful of disruptive changes to the structure of Kubernetes and feel like we should start pushing on CNCF for dedicated resources to help with product management, and push on members to contribute directly.  

I would be happy to participate in and help bootstrap a governance and project management SIG if that would be useful to the group.

**Comment by bgrant0607** [comment]
Note that we have representatives from 4 companies among the top-level approvers (OWNERS) already.

**Comment by brendandburns** [comment]
@bgrant0607 My comment was that we need to have a process for deciding to institute a process :)

e.g. I think delegating to the PM community to come up with a feature process makes sense.

I think rolling it out as required to the community (and owning the accountability for that decision) should be a formal role for a small group of owners/elders.

In the same way, the decision that we _should_ restructure the code should be done by owners/elders. The details of _how_ can be delegated to a different group of technical folks (e.g. test infra, etc)

Make sense?

**Comment by bgrant0607** [comment]
@brendandburns 

Rolling out the features process sounds like something that someone with "manager" in their title (manager, project manager, program manager, ...) would do.

Deciding that code should be restructured sounds like something a TL should do.

**Comment by brendandburns** [comment]
@bgrant0607 I don't think we can have that kind of distributed delegation for project-wide decisions.

We need a small central set of folks to guide the broad project-wide decisions.

We can't have an individual TL decide that the code needs to be restructured.

We can delegate to an individual TL once the leaders of the project decide that the code needs to be restructured. But the decision to do it, needs to be done by a small group of people who are stewards for the entire project.

What we have right now is a bunch of ad-hoc decisions made by various people who assemble support, convince people to merge proposal PRs, and otherwise make changes until something happens.

This has sort-of worked ok so far, but it leads to really poor sight lines for people in the project who are not near the center, and no real understanding of how/why certain decisions get made.

We need to have both clear high-level decision making for big project-wide decisions, as well as accountability for who made those decisions and why.

I think that we are very much lacking both right now and that opacity leads to inefficiency because people don't know what they need to do to get a big change made, and they waste effort because they don't get a clear signal that it's not going to happen. 

Likewise, it leads to frustration by various members of the community because it's sometimes difficult to understand why a broad change is being undertaken.

There are lots of examples of these things, from the decision to restructure the code base, to the decision to change the structure and organization of labels, the proposals process, etc, etc. all of these have been done in a more or less ad-hoc manner without a clear and small set of deciders who determined a central focus for what was best for the project.

I think that this worked when the project was small and everyone was in relatively tight sync, but as it has gotten more and more distributed, I don't think it is working very well.

**Comment by bgrant0607** [comment]
@brendandburns Sorry, I wasn't trying to distinguish between a group and individuals in my example, but between different roles, much as you suggested delegating the features process to the PM group.

**Comment by chrislovecnm** [comment]
@bgrant0607 / @brendandburns / @sarahnovotny / et al.

So how is the community going to come to an accord with this process?

**Comment by bgrant0607** [comment]
@chrislovecnm Will be discussed at today's hangout

**Comment by bgrant0607** [comment]
3 branches proposal: #295

Governance PR: #286

**Comment by bgrant0607** [comment]
General comment: The project is a meritocracy. People are promoted in authority based on the scope, quality, quantity, and duration of their contributions. Anyone who wants to eventually own an area needs to step up and contribute to that area first.

**Comment by bgrant0607** [comment]
Superseded by Steering Committee:
https://docs.google.com/document/d/1btvNcAKHG-3vxxpnoOk0dV5-RqyYyd198Psg11cEPVA/edit
